metadata-ingestion/src/datahub/ingestion/source/feast_repository.py

```diff
@@ -0,0 +1,368 @@
+from typing import Dict, Iterable, List, Tuple, Union
+
+from feast import (
+    BigQuerySource,
+    Entity,
+    Feature,
+    FeatureStore,
+    FeatureView,
+    FileSource,
+    KafkaSource,
+    KinesisSource,
+    OnDemandFeatureView,
+    ValueType,
+)
+from feast.data_source import DataSource, RequestDataSource
+
+import datahub.emitter.mce_builder as builder
+from datahub.configuration.common import ConfigModel
+from datahub.emitter.mce_builder import DEFAULT_ENV
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.metadata.com.linkedin.pegasus2avro.common import MLFeatureDataType
+from datahub.metadata.com.linkedin.pegasus2avro.metadata.snapshot import (
+    MLFeatureSnapshot,
+    MLFeatureTableSnapshot,
+    MLPrimaryKeySnapshot,
+)
+from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
+from datahub.metadata.schema_classes import (
+    BrowsePathsClass,
+    MLFeaturePropertiesClass,
+    MLFeatureTablePropertiesClass,
+    MLPrimaryKeyPropertiesClass,
+)
+
+_field_type_mapping: Dict[ValueType, str] = {
+    ValueType.UNKNOWN: MLFeatureDataType.UNKNOWN,
+    ValueType.BYTES: MLFeatureDataType.BYTE,
+    ValueType.STRING: MLFeatureDataType.TEXT,
+    ValueType.INT32: MLFeatureDataType.ORDINAL,
+    ValueType.INT64: MLFeatureDataType.ORDINAL,
+    ValueType.DOUBLE: MLFeatureDataType.CONTINUOUS,
+    ValueType.FLOAT: MLFeatureDataType.CONTINUOUS,
+    ValueType.BOOL: MLFeatureDataType.BINARY,
+    ValueType.UNIX_TIMESTAMP: MLFeatureDataType.TIME,
+    ValueType.BYTES_LIST: MLFeatureDataType.SEQUENCE,
+    ValueType.STRING_LIST: MLFeatureDataType.SEQUENCE,
+    ValueType.INT32_LIST: MLFeatureDataType.SEQUENCE,
+    ValueType.INT64_LIST: MLFeatureDataType.SEQUENCE,
+    ValueType.DOUBLE_LIST: MLFeatureDataType.SEQUENCE,
+    ValueType.FLOAT_LIST: MLFeatureDataType.SEQUENCE,
+    ValueType.BOOL_LIST: MLFeatureDataType.SEQUENCE,
+    ValueType.UNIX_TIMESTAMP_LIST: MLFeatureDataType.SEQUENCE,
+    ValueType.NULL: MLFeatureDataType.UNKNOWN,
+}
+
+
+class FeastRepositorySourceConfig(ConfigModel):
+    path: str
+    environment: str = DEFAULT_ENV
+
+
+class FeastRepositorySource(Source):
+    source_config: FeastRepositorySourceConfig
```

Comment:
Please move these from class-scope to instance-scope (to __init__).

Reply:
Looking for other sources, they always keep the configuration model in class-scope (eg. [MySQLConfig](https://github.com/danilopeixoto/datahub/blob/97a61c36964bf3579722d06b45f5baec110437f0/metadata-ingestion/src/datahub/ingestion/source/sql/mysql.py#L28)).

Would this change break consistency?